nestjs/fedify.module.ts

```diff
@@ -0,0 +1,45 @@
+import { DynamicModule, Module, Provider } from "@nestjs/common";
+import {
+  createFederation,
+  DocumentLoader,
+  KvStore,
+  MemoryKvStore,
+  MessageQueue,
+} from "@fedify/fedify";
+
+import { FEDIFY_FEDERATION } from "./fedify.constants.ts";
+
+export interface FedifyModuleOptions {
+  kv?: KvStore;
+  queue?: MessageQueue;
+  documentLoader?: DocumentLoader;
+  origin?: string;
+}
+
+@Module({})
+export class FedifyModule {
+  static forRoot(options: FedifyModuleOptions): DynamicModule {
```

Comment:
Hmm, I think you'd better to reuse [`FederationOptions`](https://jsr.io/@fedify/fedify/doc/~/FederationOptions) instead of declaring your own `FedifyModuleOptions`.